Jenkinsfile

```diff
@@ -108,27 +108,31 @@ dockerBuild {
                     // Test return code
                     sh './docker-run true'
                     sh 'if ./docker-run false; then false; fi'
-                    try {
-                        sh './docker-compose-run true'
-                        sh 'if ./docker-compose-run false; then false; fi'
-                    } catch (Exception error) {
-                        sh 'docker-compose logs --timestamps'
-                        throw error
-                    } finally {
-                        sh 'docker-compose down'
+                    lock('docker-compose-run') {
```

Comment:
A lock is global accros the whole CI farm. I think you want something like that, instead:
```
lock("docker-compose-run-${env.NODE_NAME}") {
```